pkg/services/store/entity/sqlstash/sql_storage_server.go

```diff
@@ -671,35 +703,43 @@ func (s *sqlEntityServer) Update(ctx context.Context, r *entity.UpdateEntityRequ
 			entityTable,
 			values,
 			map[string]any{
-				"guid": current.Guid,
+				"guid": updated.Guid,
 			},
 		)
 		if err != nil {
 			s.log.Error("error updating entity", "msg", err.Error())
 			return err
 		}
 
-		switch current.Group {
+		switch updated.Group {
 		case folder.GROUP:
-			switch current.Resource {
+			switch updated.Resource {
 			case folder.RESOURCE:
-				err = s.updateFolderTree(ctx, tx, current.Namespace)
+				err = s.updateFolderTree(ctx, tx, updated.Namespace)
 				if err != nil {
 					s.log.Error("error updating folder tree", "msg", err.Error())
 					return err
 				}
 			}
 		}
 
-		rsp.Entity = current
+		rsp.Entity = updated
 
-		return s.setLabels(ctx, tx, current.Guid, current.Labels)
+		return s.setLabels(ctx, tx, updated.Guid, updated.Labels)
 	})
 	if err != nil {
 		s.log.Error("error updating entity", "msg", err.Error())
 		rsp.Status = entity.UpdateEntityResponse_ERROR
 	}
 
+	evt := &entity.EntityWatchResponse{
+		Timestamp: time.Now().UnixMilli(),
+		Entity:    rsp.Entity,
+		Previous:  previous,
+	}
+
+	s.stream <- evt
```

Comment:
IIUC this is the part dispatching the event straightaway.
In the description it's mentioned that events are still sent by the poller but will be discarded as "too old" when running in single-server mode. Where is the discarding happening ? It kind-of feel the events will be propaged twice ? 

I am wondering if we should just forcepull in tests to keep this fast ? But not in prod 🤔

Reply:
Right now the check is in watchMatches, but it might make things clearer if we move it to inside watch itself. I have been thinking about this more and I haven't come up with a good solution yet for the multi-server issue but I agree that the simplest thing to do for now will be to merge this then we can work on that when we implement a global sequence.